Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,747 @@
+}
+
+- (id) instantiate
+{
+  NSStoryboard *sb = [NSStoryboard storyboardWithName: _storyboardName
+                                               bundle: [NSBundle mainBundle]];
+  return [sb instantiateInitialController];
+}
+
+@end
+
+@implementation GSStoryboardTransform
+
+- (instancetype) initWithData: (NSData *)data
+{
+  self = [super init];
+  if (self != nil)
+    {
+      NSXMLDocument *xml = [[NSXMLDocument alloc] initWithData: data
+                                                       options: 0
+                                                         error: NULL];
+      
+      _scenesMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _controllerMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _documentsMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      _identifierToSegueMap = [[NSMutableDictionary alloc] initWithCapacity: 10];
+      
+      [self processStoryboard: xml];
+      RELEASE(xml);
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_initialViewControllerId);
+  RELEASE(_applicationSceneId);
+  RELEASE(_scenesMap);
+  RELEASE(_controllerMap);
+  RELEASE(_documentsMap);
+  RELEASE(_identifierToSegueMap);
+  [super dealloc];
+}
+
+- (NSString *) initialViewControllerId
+{
+  return _initialViewControllerId;
+}
+
+- (NSString *) applicationSceneId
+{
+  return _applicationSceneId;
+}
+
+- (NSDictionary *) scenesMap
+{
+  return _scenesMap;
+}
+
+- (NSDictionary *) controllerMap
+{
+  return _controllerMap;
+}
+
+- (NSDictionary *) documentsMap
+{
+  return _documentsMap;
+}
+
+- (NSDictionary *) identifierToSegueMap
+{
+  return _identifierToSegueMap;
+}
+
+- (NSMapTable *) segueMapForIdentifier: (NSString *)identifier
+{
+  return [_identifierToSegueMap objectForKey: identifier];
+}
+
+- (NSXMLElement *) createCustomObjectWithId: (NSString *)ident
+                                   userLabel: (NSString *)userLabel
+                                 customClass: (NSString *)className
+{
+  NSXMLElement *customObject =
+    [[NSXMLElement alloc] initWithName: @"customObject"];
+  NSXMLNode *idValue =
+    [NSXMLNode attributeWithName: @"id"
+                     stringValue: ident];
+  NSXMLNode *usrLabel =
+    [NSXMLNode attributeWithName: @"userLabel"
+                     stringValue: userLabel];
+  NSXMLNode *customCls =
+    [NSXMLNode attributeWithName: @"customClass"
+                     stringValue: className];
+  
+  [customObject addAttribute: idValue];
+  [customObject addAttribute: usrLabel];
+  [customObject addAttribute: customCls];
+
+  AUTORELEASE(customObject);
+  
+  return customObject;
+}
+
+- (NSData *) dataForIdentifier: (NSString *)identifier
+{
+  NSString *sceneId = [_controllerMap objectForKey: identifier];
+  NSXMLDocument *xml = [_scenesMap objectForKey: sceneId];
+  [self processSegues: xml
+                forId: identifier];
+  return [xml XMLData];
+}
+
+- (NSData *) dataForSceneId: (NSString *)sceneId
```

Comment:
Looking at the difference in code between this method and the one before, I am wondering whether we should store the application scene in a different place?

Reply:
I have optimized this code somewhat.  There is now a special identifier for the "application" scene.